web/client/epics/geostory.js

```diff
@@ -461,6 +475,52 @@ export const handlePendingGeoStoryChanges = action$ =>
                     )
             )
     );
+
+/**
+ * Handle the url updates on currentPage change
+ * @param {Observable} action$ stream of actions
+ * @param {object} store redux store
+ */
+export const urlUpdateOnScroll = (action$, {getState}) =>
+    action$.ofType(UPDATE_CURRENT_PAGE)
+        .debounceTime(50) // little delay if too many UPDATE_CURRENT_PAGE actions come
+        .switchMap(({sectionId, columnId}) => {
+            if (
+                modeSelector(getState()) !== 'edit' && (!!columnId || !!sectionId)
+                && updateUrlOnScrollSelector(getState())
+            ) {
+                const urlObject = queryString.parse(window.location.href);
+                const EMPTY = 'EMPTY';
+                if (sectionId) {
+                    if (urlObject.sectionId || urlObject.columnId) {
+                        history.pushState(null, '', replace(window.location.href, /[^\&]+$/, `sectionId=${sectionId}`));
```

Comment:
> the `push` action of `connected-react-router`

does not update the URL when scrolling.

> should the query parameters start with `?` instead of `&` ?

`?` means a query that will be sent to the server. `&` means parameter in the url query. So there is a semantic difference

Reply:
> does not update the URL when scrolling.

Could you explain more about this?

I tried this actions `Rx.Observable.of(push({ search: '?param=ok' }))` and it updates only the search path.


> ? means a query that will be sent to the server. & means parameter in the url query. So there is a semantic difference

After the hash we are following the semantic provided by `connected-react-router` and if you check the state the react router location it has this structure

![image](https://user-images.githubusercontent.com/19175505/87440726-958c9f80-c5f2-11ea-8caf-a7155e0fc18a.png)

we used this before in the repo and would like to keep uniform

eg:

we extracted the search direclty from the router location in the state
https://github.com/geosolutions-it/MapStore2/blob/v2020.02.00/web/client/epics/queryparams.js#L74